main/views.py

```diff
@@ -559,9 +616,79 @@ def get(self, *args, **kwargs):
     def get_context_data(self, **kwargs):
         context = super(AccountOrders, self).get_context_data()
         context.update({'user': current_user})
+        context.update({'active_orders': self.get_active_orders()})
+        context.update({'past_orders': self.get_past_orders()})
+
         return context
 
+    def get_active_orders(self):
+        active_orders = UserUserActivity.query\
+            .filter(
+                UserUserActivity.to_user_id == current_user.id,
+                UserUserActivity.start >= date.today())\
+            .order_by(UserUserActivity.start.desc()).all()
+
+        return active_orders
+
+    def get_past_orders(self):
+        past_orders = UserUserActivity.query\
+            .filter(
+                UserUserActivity.to_user_id == current_user.id,
+                UserUserActivity.start < date.today())\
+            .order_by(UserUserActivity.start.desc()).all()
+        return past_orders
+
 app.add_url_rule(
     '/account/orders',
     view_func=AccountOrders.as_view('account_orders')
 )
+
+
+class AccountOffer(TemplateView):
+    template_name = 'user/AccountOffer.html'
+    decorators = [login_required]
+
+    def get(self, *args, **kwargs):
+        context = self.get_context_data(**kwargs)
+        return self.render_to_response(context)
+
+    def get_context_data(self, **kwargs):
+        context = super(AccountOffer, self).get_context_data()
+        context.update({'user': current_user})
+        context.update({'activity': self.get_activity(kwargs)})
+        return context
+
+    def get_activity(self, kwargs):
+        activity = UserUserActivity.query.get(kwargs.get('id'))
+        return activity
+
+
+app.add_url_rule(
+    '/account/offer/<int:id>',
+    view_func=AccountOffer.as_view('offer')
+)
+
+
+class AccountOrder(TemplateView):
+    template_name = 'user/AccountOrder.html'
+    decorators = [login_required]
+
+    def get(self, *args, **kwargs):
+        context = self.get_context_data(**kwargs)
+        return self.render_to_response(context)
+
+    def get_context_data(self, **kwargs):
+        context = super(AccountOrder, self).get_context_data()
+        context.update({'user': current_user})
+        context.update({'activity': self.get_activity(kwargs)})
+        return context
+
+    def get_activity(self, kwargs):
```

Comment:
It's not a good idea to pass generic arguments. `kwargs` doesn't tell us what this functions depends on